lib/rubocop/cop/infinum/factory_association.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require 'rubocop'
+
+module RuboCop
+  module Cop
+    module Infinum
+      # This cop looks for 'association' in factories that specify a ':factory' option
+      #
+      # @example
+      #   #bad
+      #    FactoryBot.define do
+      #      factory :book do
+      #        title { 'sth' }
+      #        association :author
+      #      end
+      #    end
+      #
+      #    FactoryBoy.define do
+      #      factory :book do
+      #        title {'sth'}
+      #        author { association :author }
+      #      end
+      #    end
+      #
+      #    #good
+      #    FactoryBot.define do
+      #      factory :book do
+      #        title { 'sth' }
+      #        author { build(:author) }
+      #      end
+      #    end
+      #
+      #    # author
+      #    FactoryBot.define do
+      #      factory :author do
+      #        # defined author attributes
+      #      end
+      #    end
+      class FactoryAssociation < ::RuboCop::Cop::Cop
```

Comment:
```suggestion
      class FactoryBotAssociation < ::RuboCop::Cop::Cop
```

not to confuse with the factory design pattern